bin/tape

```diff
@@ -41,18 +45,44 @@ if (typeof opts.ignore === 'string') {
     var matcher = ignore.createMatcher(ignoreStr);
 }
 
-opts._.forEach(function (arg) {
+var files = opts._.reduce(function (result, arg) {
     // If glob does not match, `files` will be an empty array.
     // Note: `glob.sync` may throw an error and crash the node process.
     var files = glob.sync(arg);
 
     if (!Array.isArray(files)) {
-        throw new TypeError('unknown error: glob.sync did not return an array or throw. Please report this.');
+        throw new TypeError('unknown error: glob.sync("' + arg + '") did not return an array or throw. Please report this.');
     }
 
-    files.filter(function (file) { return !matcher || !matcher.shouldIgnore(file); }).forEach(function (file) {
-        require(resolvePath(cwd, file));
-    });
+    return result.concat(files);
+}, [])
+    .filter(function (file) { return !matcher || !matcher.shouldIgnore(file); })
+    .map(function (file) { return resolvePath(cwd, file); });
+
+var promise = hasDynamicImport().then(function (hasSupport) {
+    if (hasSupport) {
+        var importOrRequire = require('./import-or-require');
+
+        return files.reduce(function (promise, file) {
+            return promise.then(function () {
+                return importOrRequire(file);
+            });
+        }, Promise.resolve());
+    } else {
+        files.forEach(require);
+    }
 });
 
+if (promise) {
```

Comment:
how could this ever be falsy?

Reply:
when there is no promise support, hasDynamicImport's thenable returns undefined. i will add a comment saying this.

Reply:
ahhh. that seems like kind of a bug in hasDynamicImport - it'd be ideal not to rely on that.

Reply:
fair enough, i rewrote it to use `has-dynamic-import`'s syntax check before checking full support and going into promise mode